lib/modules/datasource/packagist/index.ts

```diff
@@ -118,39 +118,37 @@ export class PackagistDatasource extends Datasource {
       packages,
       providersUrl,
       providersLazyUrl,
-      files,
-      includesFiles,
-      providerPackages,
+      providerIncludes,
+      includes,
+      providers,
     } = registryMeta;
-    if (files) {
-      const queue = files.map(
-        (file) => (): Promise<PackagistFile> =>
-          this.getPackagistFile(regUrl, file)
-      );
-      const resolvedFiles = await p.all(queue);
-      for (const res of resolvedFiles) {
-        for (const [name, val] of Object.entries(res.providers)) {
-          providerPackages[name] = val.sha256;
-        }
-      }
-    }
+
     const includesPackages: Record<string, ReleaseResult> = {};
-    if (includesFiles) {
-      for (const file of includesFiles) {
-        const res = await this.getPackagistFile(regUrl, file);
-        if (res.packages) {
-          for (const [key, val] of Object.entries(res.packages)) {
-            const dep = PackagistDatasource.extractDepReleases(val);
-            includesPackages[key] = dep;
-          }
+
+    const providerIncludesTasks = providerIncludes.map((file) => async () => {
+      const packagistFile = await this.getPackagistFile(regUrl, file);
+      for (const [name, val] of Object.entries(packagistFile.providers)) {
+        providers[name] = val.sha256;
+      }
+    });
+
+    const includesTasks = includes.map((file) => async () => {
+      const res = await this.getPackagistFile(regUrl, file);
+      if (res.packages) {
+        for (const [key, val] of Object.entries(res.packages)) {
+          const dep = PackagistDatasource.extractDepReleases(val);
+          includesPackages[key] = dep;
         }
       }
-    }
+    });
+
+    await p.all([...providerIncludesTasks, ...includesTasks]);
```

Comment:
```suggestion
    await Promise.all([...providerIncludesTasks, ...includesTasks]);
```
does this make a difference? 🤔 Do we've hanging requests when one request fails?

Reply:
Ah, I see your misconception. Here we're forming "tasks" of signature `() => Promise<void>` to run concurrently using `p.all()` with default threshold of 5. We use `p.all` instead of `p.map` since we actually don't perform any asynchronous "mapping" of data, doing side-effects instead (i.e. assignments). We use `p.all` instead of `Promise.all` because it allows us to limit concurrency.